undertow/src/main/java/org/wildfly/extension/undertow/deployment/JBossWebParsingDeploymentProcessor.java

```diff
@@ -74,6 +74,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 warMetaData.setJBossWebMetaData(jBossWebMetaData);
                 // if the jboss-web.xml has a distinct-name configured, then attach the value to this
                 // deployment unit
+                if(jBossWebMetaData.getValves() != null) {
+                    jBossWebMetaData.getValves().stream().forEach((valve) -> {
```

Comment:
Benchmark: How Java 8 Lambdas and Streams Can Make Your Code 5 Times Slower: http://blog.takipi.com/benchmark-how-java-8-lambdas-and-streams-can-make-your-code-5-times-slower/